### PR TITLE
Add endpoint for searching for a registration for an event

### PR DESCRIPTION
**Pull request by martinhath**
10/10 code thank you come again

Neither me nor @orhanhenrik wasn't too sure about what to do about the permission logic. Also, should we return the user object, so that we can get a nice "Velkommen, Gunnar Bovim" to the frontend?

### CODE REVIEW

File: lego/apps/events/views.py [hunk 1]
```diff
@@ -201,3 +204,32 @@ def admin_register(self, request, *args, **kwargs):
...
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        username = serializer.data['username']
+
+        try:
+            reg = self.get_queryset().get(user__username=username)
+        except Registration.DoesNotExist:
+            raise ValidationError({'error': 'No user with that username'})
```
Comment (ekmartin) [inline lego/apps/events/views.py]: not that we use the error anyway, but maybe this should say something like `User bob is not registered for the event`?

**Comment by ekmartin** [comment]
LGTM otherwise (@eirsyl?)

File: lego/apps/events/views.py [hunk 2]
```diff
@@ -201,3 +204,32 @@ def admin_register(self, request, *args, **kwargs):
...
+
+        if not get_permission_handler(Event).has_perm(request.user, 'EDIT', obj=reg.event):
+            raise PermissionDenied()
+
+        reg.presence = constants.PRESENT
+        reg.save()
+        data = PublicUserSerializer(reg.user).data
```
Comment (ekmartin) [inline lego/apps/events/views.py]: actually, @martinhath, maybe this should return the updated registration? think that would work better with the frontend and reducers and shit (we have the full name from the search result anyway)

Comment (martinhath) [reply lego/apps/events/views.py]: Ah, right! Yeah, that sounds like a good idea.

File: lego/apps/events/views.py (hunk 1 above)
Comment (martinhath) [reply lego/apps/events/views.py]: Too much typing. Close as WONTFIX

**Pull request merged**